### PR TITLE
#9768: Admin dashboard Most Viewed Products Tab only gives default attribute set's products

### PR DESCRIPTION
**Pull request by RomaKis**
Admin dashboard Most Viewed Products Tab only gives default attribute set's products

### Fixed Issues (if relevant)
1. magento/magento2#9768: Admin dashboard Most Viewed Products Tab only gives default attribute set's products

### Manual testing scenarios
1. Login to Customer Account
2. Visited Product "Minerva LumaTech™ V-Tee"
3. Login to magento admin panel
4. Check magento dashboard "Most Viewed Product" grid

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by orlangur** [comment]
@miguelbalparda please process this PR together with https://github.com/magento/magento2/pull/11597. Implementation is identical, but this PR has a test 💪

**Comment by miguelbalparda** [comment]
@RomaKis any chance you can take a look at the failing tests? If you need any help just let us know. Thanks!

**Comment by RomaKis** [comment]
@miguelbalparda, yes of course. I fixed it. Hope Jenkins won't fail again.

File: app/code/Magento/Reports/Test/Unit/Model/ResourceModel/Product/CollectionTest.php [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+namespace Magento\Reports\Test\Unit\Model\ResourceModel\Product;
+
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+
+/**
+ * Test for Magento\Reports\Model\ResourceModel\Product\Collection.
+ *
+ * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
```
Comment (orlangur) [inline app/code/Magento/Reports/Test/Unit/Model/ResourceModel/Product/CollectionTest.php]: While coupling is unavoidable you cannot suppress `ExcessiveMethodLength` in new code, please refactor test accordingly.

Use imports actively (Alt+Enter in PhpStorm) to fit method calls in one line, `getMockForAbstractClass` for interface could be replaced with `createMock`.

After that please squash changes into single commit and force push into the same branch.

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by RomaKis** [comment]
@orlangur, fixed "SuppressWarnings(PHPMD.ExcessiveMethodLength)" and squashed commits.
I cann't use createMock()(1) or createPartialMock()(2) instead of getMockForAbstractClass() for interfaces:
1. I haven't ability to setMethods().
2. I need to configure all methods which exists in interface, but i don't need all of them.

**Comment by orlangur** [comment]
@RomaKis ok, I remember I was using `getMock` for interfaces but these new methods seem to be different.

**Comment by miguelbalparda** [comment]
This is good to go and going to be merged soon, can you please backport these changes to 2.1 and 2.0 please?

**Comment by RomaKis** [comment]
@miguelbalparda, yes, I'll do it.

**Pull request merged**